src/Client/BuzzHouse/Generator/StatementGenerator.cpp

```diff
@@ -769,9 +850,9 @@ int StatementGenerator::generateAlterTable(RandomGenerator & rg, AlterTable * at
                                                                                     && !tt.isFileEngine();
                                                                             }))
                                                   .get());
-        const std::string dname = t.db ? ("d" + std::to_string(t.db->dname)) : "";
-        const std::string tname = "t" + std::to_string(t.tname);
-        const bool table_has_partitions = t.isMergeTreeFamily() && fc.tableHasPartitions<false>(dname, tname);
+        const String dname = t.db ? ("d" + std::to_string(t.db->dname)) : "";
+        const String tname = "t" + std::to_string(t.tname);
+        const bool table_has_partitions = t.isMergeTreeFamily() && fc.tableHasPartitions(false, dname, tname);
```

Comment:
This code is exactly the same as in L367-369. Not a big issue, but maybe could be refactored so the same code returns these three things?

Reply:
The code is not exactly the same